scripts/slack-music-notifier.js

```diff
@@ -1,56 +1,106 @@
-(function() {
+(function(window, $, storage) {
   'use strict';
-  var title;
+  var SPOTIFY = "spotify";
 
-  function checkSong(service, options) {
+  function checkSong(service, options, prevSong) {
+    var title;   
+    prevSong = prevSong || {};
+    
     // Delay posting song for 15 seconds to not report skipped songs
     var currentPlayTime = $(options[service].playtime).text();
     var playTimeSecs = hmsToSecondsOnly(currentPlayTime);
-    if (playTimeSecs < 15) {
+    if (!playTimeSecs || playTimeSecs < 15) {
       setTimeout(function() {
-        checkSong(service, options);
+        checkSong(service, options, prevSong);
       }, 1000);
       return;
-    }
+    }    
 
     // Pandora fades album covers, wait until the new cover is loaded
-    var opacity = $(options[service].cover).css('opacity');
-    if (opacity != 1) {
+    if (options[service].cover) {
+      var opacity = $(options[service].cover).css('opacity');
+      if (opacity != 1) {
+        setTimeout(function() {
+          checkSong(service, options, prevSong);
+        }, 1000);
+        return;
+      }
+    }
+  
+    // Check to see if this is the same as the previous song
+    title = $(options[service].title).text();
+    if (prevSong.title == title) {
```

Comment:
I dislike only checking the title. Perhaps restrict the API call to Spotify if the title varies but that, again runs into issues where songs have the same title. At least the scope is reduced.